tools/pythonpkg/duckdb/experimental/spark/sql/functions.py

```diff
@@ -1552,3 +1552,73 @@ def weekofyear(col: "ColumnOrName") -> Column:
     [Row(week=15)]
     """
     return _invoke_function_over_columns("weekofyear", col)
+
+def base64(col: "ColumnOrName") -> Column:
+    """
+    Computes the BASE64 encoding of a binary column and returns it as a string column.
+
+    .. versionadded:: 1.5.0
+
+    .. versionchanged:: 3.4.0
+        Supports Spark Connect.
+
+    Parameters
+    ----------
+    col : :class:`~pyspark.sql.Column` or str
+        target column to work on.
+
+    Returns
+    -------
+    :class:`~pyspark.sql.Column`
+        BASE64 encoding of string value.
+
+    Examples
+    --------
+    >>> df = spark.createDataFrame(["Spark", "PySpark", "Pandas API"], "STRING")
+    >>> df.select(base64("value")).show()
+    +----------------+
+    |   base64(value)|
+    +----------------+
+    |        U3Bhcms=|
+    |    UHlTcGFyaw==|
+    |UGFuZGFzIEFQSQ==|
+    +----------------+
+    """
+    if isinstance(col,str):
+        col = Column(ColumnExpression(col))
+    return _invoke_function_over_columns("base64", col.cast("BLOB"))
```

Comment:
Yea that's fine, it's just creating a cast expression, if `base64` needs a BLOB then this is fine to do here
Ideally the user already gives us a BLOB, but if they dont it would just fail and require the user to manually add the cast anyways